src/components/SelectablePill/SelectablePill.props.ts

```diff
@@ -0,0 +1,28 @@
+import { HTMLAttributes } from 'react';
+
+import { SelectablePillConfig } from './SelectablePill.styles';
+import { AvatarAppearance } from '../Avatar/types';
+
+import { IconName } from '@/utility-types/IconName';
+
+export type SelectablePillProps = {
+  text: string;
+  prefix?: string;
+  state?: 'default' | 'disabled';
+  isSelected?: boolean;
+  isInverted?: boolean;
+  tabIndex?: number;
+  custom?: SelectablePillConfig;
+  beforeComponent?: BeforeComponentProps;
+  onChange?: (state: boolean) => void;
+} & Omit<HTMLAttributes<HTMLSpanElement>, 'color'>;
+
+type BeforeComponentProps =
+  | {
```

Comment:
I think there should also be a `type', the same as the `beforeComponent' prop in the `TextInput' component. 🤔